gspread/worksheet.py

```diff
@@ -3336,3 +3338,57 @@ def add_validation(
         }
 
         return self.client.batch_update(self.spreadsheet_id, body)
+
+    def expand(
+        self,
+        top_left_range_name: str = "A1",
+        direction: TableDirection = TableDirection.table,
+        value_render_option: Optional[ValueRenderOption] = None,
+        date_time_render_option: Optional[DateTimeOption] = None,
+    ) -> List[List[str]]:
+        """Expands a cell range based on non-null adjacent cells.
+
+        Expand can be done in 3 directions defined in :class:`~gspread.utils.TableDirection`
+
+            * ``TableDirection.right``: expands only same row as starting cell on the right side up to
+              first null/empty cell
+            * ``TableDirection.down``: expands only same column as starting cell to the bottom up to
+              first null/empty cell
+            * ``TableDirection.table``: expands in both direction, first right then down.
+
+        Regardless of the direction this function always returns a matrix of data, even if it has
+        only one column.
+
+        Example::
+
+            values = [
+                ['', '',   '',   ''  , ''  , ''],
+                ['', 'B2', 'C2', 'D2', ''  , 'F2'],
+                ['', 'B3', ''  , 'D3', ''  , 'F3'],
+                ['', 'B4', 'C4', 'D4', ''  , 'F4'],
+                ['', ''  , ''  , ''  , ''  , 'F5'],
+            ]
+            >>> worksheet.expand_table(TableDirection.table, 'B2')
+            [
+                ['B2', 'C2', 'D2],
+                ['B3', ''  , 'D3'],
+                ['B4', 'C4', 'D4'],
+            ]
+
+
+        .. note::
+
+            the ``TableDirection.table`` will first look right, then look down.
+            It will not check cells located inside the table. This could lead to
+            potential empty values located in the middle of the table.
+
+        :param gspread.utils.TableDirection direction: the expand direction
+        :param str start_range: the starting cell range.
+        :rtype list(list): the resulting matrix
+        """
+
+        values = self.get(
+            value_render_option=value_render_option,
+            date_time_render_option=date_time_render_option,
```

Comment:
what happens if `get` does not return a 2D array?

for example if it returns

<pre>
- - - d
- a
- t
- - - a
</pre>

then will functions break? should we provide `pad_values=True`?